src/digital.rs

```diff
@@ -174,3 +174,22 @@ pub trait InputPin {
     /// Is the input pin low?
     fn try_is_low(&self) -> Result<bool, Self::Error>;
 }
+
+/// Single input pin that can switch from input to output mode, and vice-versa.
+pub trait IoPin<TInput, TOutput> 
+where TInput : InputPin + IoPin<TInput, TOutput>,
+      TOutput : OutputPin + IoPin<TInput, TOutput> {
+    /// Error type.
+    type Error;
+
+    /// Tries to convert this pin to input mode.
+    /// 
+    /// If the pin is already in input mode, this method should succeed.
+    fn try_switch_to_input_pin(self) -> Result<TInput, Self::Error>;
```

Comment:
Since this consumes `self`, the method should be called something like `into_`...

Reply:
Yes, thanks!